### PR TITLE
Add save button next hearing location

### PR DESCRIPTION
**Pull request by richrace**

### CODE REVIEW

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts [hunk 1]
```diff
@@ -343,4 +343,144 @@ describe("NextHearingLocation", () => {
     cy.contains("td", "Next hearing location").siblings().should("include.text", "B@1EF$1")
     cy.contains("td", "Next hearing location").siblings().contains("Editable Field").should("not.exist")
   })
+
+  context("Save correction button", () => {
+    it("Should be enabled when valid value is entered", () => {
+      cy.task("clearCourtCases")
```
Comment (richardcane) [inline cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts:349]: This looks like duplication

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts [hunk 2]
```diff
@@ -343,4 +343,144 @@ describe("NextHearingLocation", () => {
...
+      cy.get("ul.moj-sub-navigation__list").contains("Offences").click()
+      cy.get(".govuk-link").contains("Offence with HO100200 - Unrecognised Force or Station Code").click()
+      cy.get("#next-hearing-location").clear()
+      cy.get("#next-hearing-location").type("B43UY00")
+
+      cy.get("#save-next-hearing-location").should("be.enabled")
+    })
+
+    it("Should be disabled when no value is entered", () => {
+      cy.task("clearCourtCases")
```
Comment (richardcane) [inline cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts:371]: ^

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts [hunk 3]
```diff
@@ -343,4 +343,144 @@ describe("NextHearingLocation", () => {
...
+      loginAndVisit("/bichard/court-cases/0")
+
+      cy.get("ul.moj-sub-navigation__list").contains("Offences").click()
+      cy.get(".govuk-link").contains("Offence with HO100200 - Unrecognised Force or Station Code").click()
+      cy.get("#next-hearing-location").clear()
+      cy.get("#save-next-hearing-location").should("be.disabled")
+    })
+
+    it("Should be disabled when invalid value is entered", () => {
+      cy.task("clearCourtCases")
```
Comment (richardcane) [inline cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts:391]: ^

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts [hunk 4]
```diff
@@ -343,4 +343,144 @@ describe("NextHearingLocation", () => {
...
+      cy.get("ul.moj-sub-navigation__list").contains("Offences").click()
+      cy.get(".govuk-link").contains("Offence with HO100200 - Unrecognised Force or Station Code").click()
+      cy.get("#next-hearing-location").clear()
+      cy.get("#next-hearing-location").type("B01")
+
+      cy.get("#save-next-hearing-location").should("be.disabled")
+    })
+
+    it("Should be disabled when whitespace is entered", () => {
+      cy.task("clearCourtCases")
```
Comment (richardcane) [inline cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts:413]: ^

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts [hunk 5]
```diff
@@ -343,4 +343,144 @@ describe("NextHearingLocation", () => {
...
+      cy.get("ul.moj-sub-navigation__list").contains("Offences").click()
+      cy.get(".govuk-link").contains("Offence with HO100200 - Unrecognised Force or Station Code").click()
+      cy.get("#next-hearing-location").clear()
+      cy.get("#next-hearing-location").type("      ")
+
+      cy.get("#save-next-hearing-location").should("be.disabled")
+    })
+
+    it("Should be enabled when a value different from the previous one is entered", () => {
+      cy.task("clearCourtCases")
```
Comment (richardcane) [inline cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts:435]: ^

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts [hunk 6]
```diff
@@ -343,4 +343,144 @@ describe("NextHearingLocation", () => {
...
+      cy.get("#next-hearing-location").type("B21XA00")
+
+      cy.get("#next-hearing-location").clear()
+      cy.get("#next-hearing-location").type("B63AD00")
+
+      cy.get("#save-next-hearing-location").should("be.enabled")
+    })
+
+    it("Should be disabled when same value as the previous one is entered", () => {
+      cy.task("clearCourtCases")
```
Comment (richardcane) [inline cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts:460]: ^

File: src/utils/validators/isValidNextHearingLocation.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import OrganisationUnitApiResponse from "types/OrganisationUnitApiResponse"
+
+const isValidNextHearingLocation = (
```
Comment (richardcane) [inline src/utils/validators/isValidNextHearingLocation.ts:3]: Does this need a test for completeness? Some of these validation functions have been a bit misleading in the past, test could help prevent that.

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts (hunk 2 above)
Comment (tausifpatel) [reply cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts]: It has been removed

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts (hunk 3 above)
Comment (tausifpatel) [reply cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts]: It has been removed

File: cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts (hunk 4 above)
Comment (tausifpatel) [reply cypress/e2e/case-details/handle-triggers-and-exceptions/edit-fields-with-exceptions/edit-next-hearing-location.cy.ts]: It has been removed

File: src/utils/validators/isValidNextHearingLocation.ts (hunk 7 above)
Comment (tausifpatel) [reply src/utils/validators/isValidNextHearingLocation.ts:3]: Unit tests are added.

**Review by richardcane — APPROVED** [review]

**Pull request merged**